frontend/src/metabase/query_builder/components/QueryModals.jsx

```diff
@@ -75,17 +75,17 @@ class QueryModals extends React.Component {
     return modal === MODAL_TYPES.SAVE ? (
       <Modal form onClose={onCloseModal}>
         <SaveQuestionModal
-          card={this.props.card}
-          originalCard={this.props.originalCard}
+          question={this.props.question}
+          originalQuestion={this.props.originalQuestion}
           tableMetadata={this.props.tableMetadata}
           initialCollectionId={this.props.initialCollectionId}
-          onSave={async card => {
+          onSave={async question => {
             // if saving modified question, don't show "add to dashboard" modal
-            await this.props.onSave(card);
+            await this.props.onSave(question);
             onCloseModal();
           }}
-          onCreate={async card => {
-            await this.props.onCreate(card);
+          onCreate={async question => {
+            await this.props.onCreate(question);
             if (question.isDataset()) {
               onCloseModal();
               setQueryBuilderMode("view");
```

Comment:
@ranquild This is where the timing problem is - the new `onCreate` is apparently a frame or two faster, and so it leaves the editor too fast for the test.